tasks/config.yml

```diff
@@ -1,14 +1,33 @@
 ---
 
 - name: Add backup system user
-  user: name={{ rsnapshot_slave_user }} shell={{ rsnapshot_slave_user_shell }}
-        update_password=on_create
+  user:
+    name: "{{ rsnapshot_slave_user }}"
+    shell: "{{ rsnapshot_slave_user_shell }}"
+
+# Account with no password may be locked down
+# which prevent logging through SSH even with ssh key
+- name: Set password for backup user if provided
+  user:
+    name: "{{ rsnapshot_slave_user }}"
+    password: "{{ rsnapshot_slave_user_password | password_hash('sha512') }}"
+  when: rsnapshot_slave_user_password is defined
```

Comment:
`when: rsnapshot_slave_user_password` should be enough, as it is defined by default (even to False).
But this task is weird, SSH login on accounts with no password work very well with a SSH key, are you sure we need this? It depends on the SSH server configuration maybe?